### PR TITLE
support other logics, incremental, models

### PR DESCRIPTION
**Pull request by yoni206**
This PR add support for models, incremental, and combination with other logics.
For the latter, this is simply done by falling back to bit-vectors using to_nat and from_nat opeartors.

### CODE REVIEW

File: src/preprocessing/passes/bv_to_int.cpp [hunk 1]
```diff
@@ -139,13 +139,13 @@ Node BVToInt::makeBinary(Node n)
         // current has children, but we do not binarize it
         NodeBuilder<> builder(k);
         if (current.getKind() == kind::BITVECTOR_EXTRACT
-            || current.getKind() == kind::APPLY_UF)
+            || current.getKind() == kind::APPLY_UF || current.getKind() == kind::INT_TO_BITVECTOR)
```
Comment (ajreynol) [inline src/preprocessing/passes/bv_to_int.cpp:142]: The robust check here is `current.getMetaKind() == metakind::PARAMETERIZED`. This indicates if the operator should be added when constructing the node.  Do update this.

File: src/preprocessing/passes/bv_to_int.cpp [hunk 2]
```diff
@@ -242,7 +242,7 @@ Node BVToInt::eliminationPass(Node n)
           // are replaced with their eliminated counterparts.
           NodeBuilder<> builder(current.getKind());
           if (current.getKind() == kind::BITVECTOR_EXTRACT
-              || current.getKind() == kind::APPLY_UF)
+              || current.getKind() == kind::APPLY_UF || current.getKind() == kind::INT_TO_BITVECTOR)
```
Comment (ajreynol) [inline src/preprocessing/passes/bv_to_int.cpp:245]: Ditto.

File: src/preprocessing/passes/bv_to_int.cpp [hunk 3]
```diff
@@ -736,38 +764,57 @@ Node BVToInt::bvToInt(Node n)
...
+                  Node adjusted_child = d_nm->mkNode(intToBVOp, translated_child);
+                  adjusted_children.push_back(adjusted_child);
+                }
               }
-              else {
-                d_bvToIntCache[current] =
-                    d_nm->mkNode(oldKind, translated_children);
+
+              NodeBuilder<> builder(oldKind);
+              if (oldKind == kind::BITVECTOR_EXTRACT
```
Comment (ajreynol) [inline src/preprocessing/passes/bv_to_int.cpp:789]: Ditto.

**Comment by yoni206** [comment]
10x. Fixed according to your comments.

**Pull request merged**